modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/action/test/EditContentDashboardConfigurationMVCRenderCommandTest.java

```diff
@@ -0,0 +1,314 @@
+
+		Assert.assertTrue(
+			availableVocabularyNames.contains(
+				new KeyValuePair(
+					_stageAssetVocabulary.getName(),
+					_stageAssetVocabulary.getTitle(_locale))));
+	}
+
+	@Test
+	public void testGetCurrentVocabularyNames() throws Exception {
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequest(
+				_audienceAssetVocabulary.getName(),
+				_stageAssetVocabulary.getName());
+
+		MockLiferayPortletRenderResponse mockLiferayPortletRenderResponse =
+			new MockLiferayPortletRenderResponse();
+
+		_mvcRenderCommand.render(
+			mockLiferayPortletRenderRequest, mockLiferayPortletRenderResponse);
+
+		List<KeyValuePair> currentVocabularyNames = ReflectionTestUtil.invoke(
+			mockLiferayPortletRenderRequest.getAttribute(
+				"CONTENT_DASHBOARD_ADMIN_CONFIGURATION_DISPLAY_CONTEXT"),
+			"getCurrentVocabularyNames", new Class<?>[0]);
+
+		Assert.assertEquals(
+			currentVocabularyNames.toString(), 2,
+			currentVocabularyNames.size());
+
+		Assert.assertEquals(
+			currentVocabularyNames.get(0),
+			new KeyValuePair(
+				_audienceAssetVocabulary.getName(),
+				_audienceAssetVocabulary.getTitle(_locale)));
+
+		Assert.assertEquals(
+			currentVocabularyNames.get(1),
+			new KeyValuePair(
+				_stageAssetVocabulary.getName(),
+				_stageAssetVocabulary.getTitle(_locale)));
+	}
+
+	@Test
+	public void testGetCurrentVocabularyNamesWithAudienceAssetVocabulary()
+		throws Exception {
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequest(
+				_audienceAssetVocabulary.getName());
+
+		MockLiferayPortletRenderResponse mockLiferayPortletRenderResponse =
+			new MockLiferayPortletRenderResponse();
+
+		_mvcRenderCommand.render(
+			mockLiferayPortletRenderRequest, mockLiferayPortletRenderResponse);
+
+		List<KeyValuePair> currentVocabularyNames = ReflectionTestUtil.invoke(
+			mockLiferayPortletRenderRequest.getAttribute(
+				"CONTENT_DASHBOARD_ADMIN_CONFIGURATION_DISPLAY_CONTEXT"),
+			"getCurrentVocabularyNames", new Class<?>[0]);
+
+		Assert.assertEquals(
+			currentVocabularyNames.toString(), 1,
+			currentVocabularyNames.size());
+
+		Assert.assertEquals(
+			currentVocabularyNames.get(0),
+			new KeyValuePair(
+				_audienceAssetVocabulary.getName(),
+				_audienceAssetVocabulary.getTitle(_locale)));
+	}
+
+	@Test
+	public void testGetRedirect() throws Exception {
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequest();
+
+		MockLiferayPortletRenderResponse mockLiferayPortletRenderResponse =
+			new MockLiferayPortletRenderResponse();
+
+		_mvcRenderCommand.render(
+			mockLiferayPortletRenderRequest, mockLiferayPortletRenderResponse);
+
+		LiferayPortletURL redirect = ReflectionTestUtil.invoke(
+			mockLiferayPortletRenderRequest.getAttribute(
+				"CONTENT_DASHBOARD_ADMIN_CONFIGURATION_DISPLAY_CONTEXT"),
+			"getRedirect", new Class<?>[0]);
+
+		Assert.assertEquals(
```

Comment:
The same I comment in the getActionURL test, since we can't assure we have a valid redirect, I think this code should be tested functionally.